### PR TITLE
Remove URL escaping in link labels

### LINKED ISSUE #4: Link text are getting escaped like this http\://localhost
### Subject of the issue

Following markdown gets replaced with an escaped link text.

```md
[http://localhost]: http://localhost "Your localhost webserver running on http"

# My document title

This is my input markdown using a link reference [http://localhost][].
```

Which in turn is translated to 

```md
[http://localhost]: http://localhost "Your localhost webserver running on http"

# My document title

This is my input markdown using a link reference [http\://localhost][http://localhost].
```

### Your environment

*   **OS**: MacOS
*   **Packages**: remark 13.0.0, remark-gfm 1.0.0
*   **Env**: node 15.0.3

### Steps to reproduce

https://codesandbox.io/s/remark-debug-forked-ocyxp?file=/src/index.js

### Expected behavior

The result should be like

```md
[http://localhost]: http://localhost "Your localhost webserver running on http"

# My document title

This is my input markdown using a link reference [http://localhost][http://localhost] and some more text.
```

### Actual behavior

```md
[http://localhost]: http://localhost "Your localhost webserver running on http"

# My document title

This is my input markdown using a link reference [http\://localhost][http://localhost] and some more text.
```

### PR DESCRIPTION
**Pull request by marcofranssen**
resolves https://github.com/remarkjs/remark-gfm/issues/4

### CODE REVIEW

**Comment by wooorm** [comment]
Thanks for working on this! Could you also add tests?

**Comment by marcofranssen** [comment]
@wooorm added the testcase. It works. Could you let me know when `remark-gfm` gets a new release?

File: test.js [hunk 1]
```diff
@@ -115,10 +115,245 @@ test('markdown -> mdast', function (t) {
...
+        ],
+        position: {
+          start: {line: 1, column: 1, offset: 0},
+          end: {line: 4, column: 72, offset: 204}
+        }
+      },
+      {extensions: [autolinkLiterals.toMarkdown]}
+    ),
+    `[http://localhost]: http://localhost "Open example on localhost"
+
```
Comment (marcofranssen) [inline test.js:349]: Only thing that still disturbes me a bit is the additional whiteline here, but that is not related to the current issue I resolved.

Comment (wooorm) [reply test.js:349]: I think it’s a good default but it’s configurable: https://github.com/syntax-tree/mdast-util-to-markdown#optionstightdefinitions

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
Released! https://github.com/syntax-tree/mdast-util-gfm-autolink-literal/releases/tag/0.1.2

**Pull request merged**